### ISSUE TITLE
[Emails] Bug: change related object to task when specific action is for a person

### ISSUE THREAD

**Issue by pbanaszkiewicz** [carpentries/amy#2719]
There are logic issues when email action is specific for a person (e.g. instructor confirmed for workshop) and that person is edited (e.g. email updated).

Switch Instructor Confirmed for Workshop to use Tasks instead. Perhaps switch some other actions, too (instructor training completed not badged, instructor badge awarded).

**Comment by maneesha** [comment]
@pbanaszkiewicz We are still seeing this issue.  See notes in our agenda document for examples.

**Comment by pbanaszkiewicz** [comment]
@maneesha Tested with person https://test-amy2.carpentries.org/workshops/person/108/.

1. I updated their email for the first time. A lot of new email actions for intructor confirmed for workshop appeared.
2. I updated their email again. A lot of email actions for instructor confirmed for workshop appeared, but they were updated - not created as new.
3. I cancelled all of those emails and updated person's email again. This time no new/updated emails for instructor confirmed for workshop appeared.

All these are expected. Since the changes I made in v4.4 touched related object for instructor confirmed email action, the first time you change person who had previously some instructor confirmed email actions - this time new ones (duplicated) would appear. Any future changes to that person would not trigger those actions.

As I checked production AMY, there were not many instructors with that email in the queue, so I assumed that solution, despite allowing for duplicated emails in some cases, was necessary and better for the future.
